
---------------
bool success
string type
string[] label
spot_data[] spot
bool end
